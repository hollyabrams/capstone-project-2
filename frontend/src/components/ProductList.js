import { useState, useEffect } from "react";
import SearchForm from "../forms/SearchForm";
import ModeApi from "../api";
import ProductCard from "./ProductCard";

/** Show page with list of products.
 *
 * On mount, loads products from API.
 * Re-loads filtered products on submit from search form.
 *
 * This is routed to at /products
 *
 * Routes -> { ProductCard, SearchForm }
 */

const ProductList = () => {
  const [products, setProducts] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(function getProductsOnMount() {
    console.debug('ProductList useEffect getProductsOnMount');
    search();
  }, []);

  /** Triggered by search form submit; reloads companies. */
  async function search(name) {
    let products = await ModeApi.getProducts(name);
    setProducts(products);
    setIsLoading(false);
  }

  if (isLoading) {
    return (
      <p>Loading...</p>
    )
  }

  return (
    <div className="container xl:max-w-screen-xl mx-auto py-12 px-20">
      <SearchForm searchFor={search} />
      {products.length ? (
        <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-8 mb-8">
          {products.map(p => (
            <ProductCard
              key={p.id}
              product={p}
            />
          ))}
        </div>
      ) : (
        <p className="text-center text-xl text-gray-500">Sorry, no results were found!</p>
      )}
    </div>
  );
};

export default ProductList;