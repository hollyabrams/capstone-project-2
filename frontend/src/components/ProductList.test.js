// ProductList.test.js
import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import ProductList from './ProductList';

test('ProductList renders without crashing', () => {
  render(
    <Router>
      <ProductList />
    </Router>
  );
});

test('ProductList matches snapshot', () => {
  const { asFragment } = render(
    <Router>
      <ProductList />
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});
