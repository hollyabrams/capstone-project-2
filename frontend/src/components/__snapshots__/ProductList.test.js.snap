// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList matches snapshot 1`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;
